### PR TITLE
Reduce boilerplate code around handling gRPC error

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by knanao** [comment]
awesome:D
/lgtm

File: pkg/app/server/grpcapi/grpcapi.go [hunk 1]
```diff
@@ -191,6 +191,22 @@ func getEncriptionKey(se *model.Piped_SecretEncryption) ([]byte, error) {
 	}
 }
 
+func gRPCErrorForEntityOperation(err error, op string, a ...interface{}) error {
+	if err == nil {
+		return nil
+	}
+	switch err {
+	case datastore.ErrNotFound:
```
Comment (nakabonne) [inline pkg/app/server/grpcapi/grpcapi.go:199]: nit: it's a little bit weird to evaluate err twice, but it's up to you.
```suggestion
	switch err {
	case nil:
	        return nil
	case datastore.ErrNotFound:
```

File: pkg/app/server/grpcapi/grpcapi.go [hunk 2]
```diff
@@ -191,6 +191,22 @@ func getEncriptionKey(se *model.Piped_SecretEncryption) ([]byte, error) {
 	}
 }
 
+func gRPCErrorForEntityOperation(err error, op string, a ...interface{}) error {
```
Comment (nakabonne) [inline pkg/app/server/grpcapi/grpcapi.go:194]: Well, is it really needed to pass `a ...interface{}`?
I feel like it's enough that the caller side just does:
```
gRPCErrorForEntityOperation(err, fmt.Sprintf("add event %s", id))
```

Then:
```
return status.Error(codes.Internal, fmt.Sprintf("Failed to %s", op))
```

`add event 198923jfnk-ao23enfaoe-3fafej` is obviously an operation.

Comment (nghialv) [reply pkg/app/server/grpcapi/grpcapi.go:194]: I feel you. It is just to help the caller a bit but I have no strong opinions on that. 

I can change that method if you guys want or rename it to be more clear like below. 👍 

```
func gRPCErrorForEntityOperation(err error, opFormat string, opValue ...interface{}) error
```

@khanhtc1202 @knanao Nice to have your thoughts as well.

Comment (khanhtc1202) [reply pkg/app/server/grpcapi/grpcapi.go:194]: Well, I think this function is just for building the gRPC error code, others should be built outside and pass throw the function parameters. So, how about rename it as
```golang
func gRPCErrorForEntityOperation(err error, msg string) error
```
just like they do for the `status.Error`

Screen Shot 2022-02-02 at 13 38 16

Comment (khanhtc1202) [reply pkg/app/server/grpcapi/grpcapi.go:194]: The implementation will be error message wrapper like
```golang
case datastore.ErrNotFound:
	return status.Error(codes.NotFound, fmt.Sprintf("Entity not found: %s", msg))
```

Comment (nghialv) [reply pkg/app/server/grpcapi/grpcapi.go:194]: Ok. So let me change to being like that. 👍 
FYI, they have `status.Errorf` as well. https://github.com/grpc/grpc-go/blob/v1.44.0/status/status.go#L63

Comment (khanhtc1202) [reply pkg/app/server/grpcapi/grpcapi.go:194]: oh nice, then 
```golang
func gRPCEntityOperationErrorf(err error, format string, a ...interface{}) error
```
is good to me as well. I still think that this function is only for wrapping error code, but it's up to you 😉

Comment (nghialv) [reply pkg/app/server/grpcapi/grpcapi.go]: Thank you for your comments. Fixed it. PTAL.
https://github.com/pipe-cd/pipecd/pull/3170/commits/b28d93378472b7d039a421309767d024d8e723fe

File: pkg/app/server/grpcapi/grpcapi.go [hunk 3]
```diff
@@ -191,6 +191,21 @@ func getEncriptionKey(se *model.Piped_SecretEncryption) ([]byte, error) {
 	}
 }
 
+func gRPCErrorForEntityOperation(err error, msg string) error {
```
Comment (khanhtc1202) [inline pkg/app/server/grpcapi/grpcapi.go:194]: nits
```suggestion
func gRPCEntityOperationError(err error, msg string) error {
```
this name is more natural.

**Comment by khanhtc1202** [comment]
small nits, but it's LGTM
/lgtm

File: pkg/app/server/grpcapi/grpcapi.go (hunk 3 above)
Comment (nghialv) [reply pkg/app/server/grpcapi/grpcapi.go]: Nice. Applied it! 👍

**Comment by khanhtc1202** [comment]
Nice patch
/lgtm

**Comment by nakabonne** [comment]
Nice!
/approve

**Pull request merged**